sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,281 @@
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            if self.get_tag("scitype:y")=="univariate":
+                guaranteed to have a single column
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+            Required (non-optional) here.
+        X : pd.DataFrame, optional (default=None)
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        del fh  # avoid being detected as unused by ``vulture`` like tools
+
+        endogenous_actuals = self.convert_dataframe_to_timeseries(y)
+        unknown_exogenous, known_exogenous = self.convert_exogenous_dataset(X)
+        self._forecaster = self._create_forecaster()
+        self._forecaster.fit(
+            endogenous_actuals,
+            past_covariates=unknown_exogenous,
+            future_covariates=known_exogenous,
+        )
+
+        return self
+
+    def _predict(
+        self: "_DartsAdapter",
+        fh: Optional[ForecastingHorizon],
+        X: Optional[pd.DataFrame],
+    ):
+        """Forecast time series at future horizon.
+
+        private _predict containing the core logic, called from predict
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous time series
+
+        Returns
+        -------
+        y_pred : pd.DataFrame
+            Point predictions
+        """
+        unknown_exogenous, known_exogenous = self.convert_exogenous_dataset(X)
+
+        maximum_forecast_horizon = fh.to_relative(self.cutoff)[-1]
+        endogenous_point_predictions = self._forecaster.predict(
+            maximum_forecast_horizon,
+            past_covariates=unknown_exogenous,
+            future_covariates=known_exogenous,
+            num_samples=1,
+        ).pd_dataframe()
+
+        if X is not None and len(X.columns) == len(
+            endogenous_point_predictions.columns
+        ):
+            endogenous_point_predictions.columns = X.columns
+        else:
+            endogenous_point_predictions.columns = range(
+                len(endogenous_point_predictions.columns)
+            )
+        return endogenous_point_predictions
+
+    # todo 0.22.0 - switch legacy_interface default to False
+    # todo 0.23.0 - remove legacy_interface arg
+    def _predict_quantiles(
```

Comment:
We should update this part of code. Please remove these comments, the deprecation argument, and update with new format.